src/node/internal/internal_http_incoming.ts

```diff
@@ -113,27 +163,183 @@ export class IncomingMessage extends Readable implements _IncomingMessage {
+      // Array header -- only Set-Cookie at the moment
+      if (dest['set-cookie'] !== undefined) {
+        dest['set-cookie'].push(value);
+      } else {
+        dest['set-cookie'] = [value];
+      }
+    } else if (this.joinDuplicateHeaders) {
+      // RFC 9110 https://www.rfc-editor.org/rfc/rfc9110#section-5.2
+      // https://github.com/nodejs/node/issues/45699
+      // allow authorization multiple fields
+      // Make a delimited list
+      if (dest[field] === undefined) {
+        dest[field] = value;
+      } else {
+        // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+        dest[field] += ', ' + value;
+      }
+    } else if (dest[field] === undefined) {
+      // Drop duplicates
+      dest[field] = value;
+    }
+  }
+
+  _addHeaderLines(headers: string[] | null, n: number): void {
+    if (Array.isArray(headers)) {
+      let dest;
+      if (this.complete) {
+        this.rawTrailers = headers;
+        this[kTrailersCount] = n;
+        dest = this[kTrailers];
+      } else {
+        this.rawHeaders = headers;
+        this[kHeadersCount] = n;
+        dest = this[kHeaders];
+      }
+
+      if (dest) {
+        for (let i = 0; i < n; i += 2) {
+          this._addHeaderLine(
+            headers[i] as string,
+            headers[i + 1] as string,
+            dest
+          );
+        }
+      }
+    }
+  }
+
+  get headers(): Record<string, string | string[] | undefined> {
+    if (!this[kHeaders]) {
+      this[kHeaders] = {};
+
+      const src = this.rawHeaders;
+      const dst = this[kHeaders];
+
+      for (let n = 0; n < this[kHeadersCount]; n += 2) {
+        this._addHeaderLine(src[n] as string, src[n + 1] as string, dst);
+      }
+    }
+    return this[kHeaders];
+  }
+
+  set headers(val: IncomingHttpHeaders) {
+    this[kHeaders] = val;
   }
 
-  // @ts-expect-error TS2416 Type inconsistency
-  get headersDistinct(): Record<string, string> {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument,@typescript-eslint/no-explicit-any
-    return Object.fromEntries(this.#response.headers as any);
+  get headersDistinct(): Record<string, string[]> {
+    if (!this[kHeadersDistinct]) {
+      this[kHeadersDistinct] = {};
+
+      const src = this.rawHeaders;
+      const dst = this[kHeadersDistinct];
+
+      for (let n = 0; n < this[kHeadersCount]; n += 2) {
+        this._addHeaderLineDistinct(
+          src[n] as string,
+          src[n + 1] as string,
+          dst
+        );
+      }
+    }
+    return this[kHeadersDistinct];
   }
 
-  // @ts-expect-error TS2416 Type inconsistency
-  get trailers(): Record<string, unknown> {
-    // Not supported.
-    return {};
+  set headersDistinct(val: Record<string, string[]>) {
+    this[kHeadersDistinct] = val;
   }
 
-  // @ts-expect-error TS2416 Type inconsistency
-  get trailersDistinct(): Record<string, unknown> {
-    // Not supported.
-    return {};
+  get trailers(): Record<string, string | undefined> {
```

Comment:
Not blocking but We know that we don't support trailers so I'm not sure there's much value in implementing this at all.

Reply:
I'm closing this for now, but LMK if this is a deal-breaker for you (by unresolving)